framework/mesh/mesh_matrix3x3.h

```diff
@@ -68,219 +57,177 @@ struct Matrix3x3
   }
 
   /// Set column j using a vector.
-  void SetColJVec(int j, Vector3 vec)
+  void SetColJVec(int j, const Vector3& vec)
   {
-    vals[j + 3 * 0] = vec.x;
-    vals[j + 3 * 1] = vec.y;
-    vals[j + 3 * 2] = vec.z;
+    vals[0 * 3 + j] = vec.x;
+    vals[1 * 3 + j] = vec.y;
+    vals[2 * 3 + j] = vec.z;
   }
 
   /// Sets the diagonal of the matrix.
   void SetDiagonalVec(double a00, double a11, double a22)
   {
-    vals[0 + 3 * 0] = a00;
-    vals[1 + 3 * 1] = a11;
-    vals[2 + 3 * 2] = a22;
+    vals[0 * 3 + 0] = a00;
+    vals[1 * 3 + 1] = a11;
+    vals[2 * 3 + 2] = a22;
   }
 
   /// Get the determinant using specified row [default:0].
-  double Det(int row = 0)
+  [[nodiscard]] double Det(int row = 0) const
   {
     double det = 0.0;
-
     int sign = -1;
+
     for (int j = 0; j < 3; ++j)
     {
-      int k = j + 3 * row;
       sign *= -1;
-
       det += sign * GetIJ(row, j) * MinorIJ(row, j);
     }
 
     return det;
   }
 
   /// Get the minor value associated with row ir and column jr.
-  double MinorIJ(int ir, int jr)
+  [[nodiscard]] double MinorIJ(int ir, int jr) const
   {
-    double a[] = {0.0, 0.0, 0.0, 0.0};
+    std::array<double, 4> a{};
+    int k = 0;
 
-    int k = -1;
     for (int i = 0; i < 3; ++i)
     {
       if (i == ir)
         continue;
-
       for (int j = 0; j < 3; ++j)
       {
         if (j == jr)
           continue;
-
-        k++;
-        int kr = j + 3 * i;
-        a[k] = vals[kr];
+        a[k++] = vals[i * 3 + j];
       }
     }
 
-    return a[0] * a[3] - a[2] * a[1];
+    return a[0] * a[3] - a[1] * a[2];
   }
 
   /// Compute the matrix transpose.
-  Matrix3x3 Transpose()
+  [[nodiscard]] Matrix3x3 Transpose() const
   {
-    Matrix3x3 oM;
+    Matrix3x3 result;
     for (int i = 0; i < 3; ++i)
-    {
       for (int j = 0; j < 3; ++j)
-      {
-        int kO = j + 3 * i;
-        int kI = i + 3 * j;
+        result.vals[j * 3 + i] = vals[i * 3 + j]; // Swap row and column
 
-        oM.vals[kO] = vals[kI];
-      }
-    }
-    return oM;
+    return result;
   }
 
-  /// Compute the matrix transpose.
-  Matrix3x3 Inverse()
+  /// Compute the matrix inverse.
+  [[nodiscard]] Matrix3x3 Inverse() const
   {
-    Matrix3x3 oM;
-    Matrix3x3 oMT;
+    Matrix3x3 oM;  // Matrix of minors
+    Matrix3x3 oMT; // Transpose of cofactors
 
     // Compute matrix of minors
     for (int i = 0; i < 3; ++i)
-    {
       for (int j = 0; j < 3; ++j)
-      {
         oM.SetIJ(i, j, MinorIJ(i, j));
-      }
-    }
 
     // Compute matrix of cofactors
     int sign = -1;
     for (int i = 0; i < 3; ++i)
-    {
       for (int j = 0; j < 3; ++j)
       {
         sign *= -1;
         oM.SetIJ(i, j, oM.GetIJ(i, j) * sign);
       }
-    }
 
     // Compute the transpose
     oMT = oM.Transpose();
 
     // Get determinant
     double det = Det();
 
-    return oMT * (1.0 / det);
-  }
-
-  /// Outputs the matrix to a stream.
-  friend std::ostream& operator<<(std::ostream& out, Matrix3x3& inM)
```

Comment:
This was really useful for debugging. One could do:
```
std::cerr << mat << std::endl;
```
Reconsider the removal...

Reply:
but isn't PrintS already doing this?

Reply:
If you want users/developers do: 
```
std::cerr << mat.PrintStr() << std::endl;
```
then, the removal is ok.

Reply:
I have remove all instances of `PrintS()` and replaced them by `PrintStr()` so that the deprecated method in mesh_vector can be deleted. I changed the `PrintS()` method of mesh_matrix3x3 by `PrintStr()` for consistency but `PrintS()` of mesh_matrix3x3 is unused, so no effect.